boards/p1am_100/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+            PullUpInput: BaseSlaveAck
+        }
+    }
+    PA06 {
+        name: d20,
+        aliases: {
+            AlternateB: A5
+        }
+    }
+    PA07 {
+        name: d21,
+        aliases: {
+            AlternateB: A6,
+            AlternateG: I2sSerialData
+        }
+    }
+    PA08 {
+        name: d11,
+        aliases: {
+            AlternateC: Sda
+        }
+    }
+    PA09 {
+        name: d12,
+        aliases: {
+            AlternateC: Scl
+        }
+    }
+    PA10 {
+        name: d2,
+        aliases: {
+            AlternateG: I2sSerialClock
+        }
+    }
+    PA11 { name: d3 }
+    PA12 {
+        name: pa12,
+        aliases: {
+            AlternateC: SdSdo
+        }
+    }
+    PA13 {
+        name: pa13,
+        aliases: {
+            AlternateC: SdSck
+        }
+    }
+    PA14 {
+        name: pa14,
+        aliases: {
+            PushPullOutput: SdSlaveSelect
+        }
+    }
+    PA15 {
+        name: pa15,
+        aliases: {
+            AlternateC: SdSdi
+        }
+    }
+    PA16 {
+        name: d8,
+        aliases: {
+            AlternateC: Spi0Mosi
+        }
+    }
+    PA17 {
+        name: d9,
+        aliases: {
+            AlternateC: Spi0Sck
+        }
+    }
+    PA18 {
+        name: pa18,
+        aliases: {
+            /// Host Enable, drive high to switch into USB host mode
+            #[cfg(feature = "usb")]
+            PushPullOutput: UsbId
+        }
+    }
+    PA19 {
+        name: d10,
+        aliases: {
+            AlternateC: Spi0Miso
+        }
+    }
+    PA20 { name: d6 }
+    PA21 { name: d7 }
+    PA22 { name: d0 }
+    PA23 { name: d1 }
+    PA24 {
+        name: usb_dm,
+        aliases: {
+            #[cfg(feature = "usb")]
+            AlternateG: UsbDm
+        }
+    }
+    PA25 {
+        name: usb_dp,
+        aliases: {
+            #[cfg(feature = "usb")]
+            AlternateG: UsbDp
+        }
+    }
+    PA27 {
+        name: pa27,
+        aliases: {
+            PullUpInput: SdCardDetect
+        }
+    }
+    PA28 {
+        name: switch,
+        aliases: {
+            PullUpInput: Switch
+        }
+    }
+    PB02 {
+        name: d16,
+        aliases: {
+            AlternateB: A1
+        }
+    }
+    PB03 {
+        name: d17,
+        aliases: {
+            AlternateB: A2
+        }
+    }
+    PB08 {
+        name: led,
+        aliases: {
+            PushPullOutput: Led
+        }
+    }
+    PB09 {
+        name: pb09,
+        aliases: {
+            PushPullOutput: BaseEnable,
+            AlternateB: AdcBattery
+        }
+    }
+    PB10 { name: d4 }
+    PB11 { name: d5 }
+    PB22 {
+        name: d14,
+        aliases: {
+            AlternateD: UartTx
+        }
+    }
+    PB23 {
+        name: d13,
+        aliases: {
+            AlternateD: UartRx
+        }
+    }
+);
+
+const BASE_CONTROLLER_FREQ: Hertz = Hertz(1000000);
+// FIXME: const BASE_CONTROLLER_SPI_MODE: dyn spi::Mode = &spi::MODE_2;
```

Comment:
I see your confusion here. There are two different types of "mode". There's the SPI mode, which is a combination of CPOL and CPHA. Then there's the operating mode, of which there are three, `Master`, `MasterHWSS` and `Slave`. Check the documentation for the `Mode` trait for more details. If the documentation is lacking (which it probably is), let me know what you don't understand and I can expand.

Reply:
Ahh, I need to explicitly use `embedded_hal::spi::Mode` as the type instead of `hal::spi::Mode`, even though the constant is `hal::spi::MODE_2`. Boy, that's confusing.